public/app/features/alerting/unified/components/DataSourceSelect.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { FC, useEffect, useState } from 'react';
+import { Select, InputControl } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+
+import { getRulesDataSources } from '../utils/datasource';
+import { getAllDataSources } from '../utils/config';
+import { ALERT_TYPE } from './rule-editor/AlertTypeSection';
+import { AlertRuleFormMethods } from './rule-editor/AlertRuleForm';
+
+interface Props {
+  className?: string;
+  control?: AlertRuleFormMethods['control'];
+  alertType?: ALERT_TYPE;
+  onChange?: (item: SelectableValue) => void;
+  value?: string;
+}
+
+const DataSourceSelect: FC<Props> = ({ alertType, className, control, onChange, value }) => {
+  const dataSourceOptions = useDatasourceSelectOptions(alertType);
+
+  if (control) {
+    return (
+      <InputControl
+        className={className}
+        as={Select}
+        control={control}
+        name={'dataSource'}
+        options={dataSourceOptions}
+      />
+    );
+  } else if (onChange) {
+    return <Select className={className} options={dataSourceOptions} onChange={onChange} value={value} />;
+  } else {
+    return null;
+  }
+};
+
+const useDatasourceSelectOptions = (alertType?: ALERT_TYPE) => {
+  const [datasourceOptions, setDataSourceOptions] = useState<SelectableValue[]>([]);
+
+  useEffect(() => {
+    let options = [] as ReturnType<typeof getAllDataSources>;
```

Comment:
will need an option for `grafana`, like the `AlertManagerSelect` has

Reply:
We have a component for this alrady, DataSourcePicker

Reply:
We should not add another one, if you need more control over filtering what data sources to show in list you can add that to that component

Reply:
I added some new filters for the `DataSourcePicker` component, allowing to filter by whether it supports alerting also by data source type.